kedro/io/core.py

```diff
@@ -234,6 +236,8 @@ def _to_str(obj: Any, is_root: bool = False) -> str:
             formatted like Dataset(key=value).
             2. Dictionaries have the keys alphabetically sorted recursively.
             3. None values are not shown.
+
+            Will be removed in 0.20.0 release.
```

Comment:
Is this a deprecation meant to be seen by users? It should at least be mentioned in the release notes, but usually we throw a deprecation warning as well, see: https://github.com/kedro-org/kedro/blob/main/kedro/framework/cli/utils.py#L198-L200

Reply:
This comment refers to a private method, so it's not for users to see but for us. I replaced it with TODO for clarity. The aim is to remove the current `__str__` implementation, so `__repr__` is used by default, so it's not a deprecation but an update.